### PR TITLE
[DPP-1278] Handle almost all the remaining 'um-for-hub' todos

### PR DESCRIPTION
**Pull request by pbatko-da**
Easy to review commit by commit

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto [hunk 1]
```diff
@@ -267,7 +267,7 @@ message GrantUserRightsRequest {
     string user_id = 1;
 
     // The rights to grant.
-    // Required
+    // Optional
```
Comment (pbatko-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto:270]: Looks like we've always allowed to grant an empty list of rights and so this field is really an optional one. Reflecting that in the proto comment

Comment (mziolekda) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto:270]: Is that really optional? What's the use of a grant rights call where this field is not populated?

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala [hunk 2]
```diff
@@ -218,7 +217,7 @@ private[apiserver] final class ApiPartyManagementService private (
             "update_mask",
           )
           displayNameO <- FieldValidations.optionalString(partyDetails.displayName)(Right(_))
-          partyRecord = ParticipantParty.PartyDetails(
+          partyRecord = domain.ParticipantPartyDetails(
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala:220]:
```suggestion
          partyRecord = ParticipantPartyDetails(
```

File: ledger/participant-integration-api/src/main/scala/platform/localstore/PersistentPartyRecordStore.scala [hunk 3]
```diff
@@ -52,9 +52,9 @@ class PersistentPartyRecordStore(
 
   private val logger = ContextualizedLogger.get(getClass)
 
-  override def createPartyRecord(partyRecord: domain.ParticipantParty.PartyRecord)(implicit
+  override def createPartyRecord(partyRecord: domain.PartyRecord)(implicit
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/localstore/PersistentPartyRecordStore.scala:55]: Suggest to use consistently either `domain.PartyRecord` or `PartyRecord` but not both in a single file

File: ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/ResourceVersionOpsBackend.scala [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.localstore
+
+import java.sql.Connection
+
+trait ResourceVersionOpsBackend {
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/ResourceVersionOpsBackend.scala:8]: That class is in a different category to all the other back-ends. I suggest to call it just `ResourceVersionOps`

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto (hunk 1 above)
Comment (pbatko-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto:270]: It's how the implementation works. We never enforced the rights in the request to be non-empty.
We could change it, but then it's a breaking change in the LAPI

File: ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/ResourceVersionOpsBackend.scala (hunk 4 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/ResourceVersionOpsBackend.scala:8]: Ok

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala:220]: Updating to use `ParticipantPartyDetails`

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto (hunk 1 above)
Comment (mziolekda) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto:270]: ok

**Review by mziolekda — APPROVED** [review]
Added one more rename suggestion. I don't feel strongly about it though
Approving

**Pull request merged**